OpenRA.Mods.Cnc/FileSystem/MegFile.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using OpenRA.FileSystem;
+using OpenRA.Primitives;
+
+namespace OpenRA.Mods.Cnc.FileSystem
+{
+	public class MegV3Loader : IPackageLoader
+	{
+		public bool TryParsePackage(Stream s, string filename, OpenRA.FileSystem.FileSystem context, out IReadOnlyPackage package)
+		{
+			if (!filename.EndsWith(".meg", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Please don't rely on an extension check here. Read the first 8 bytes and check that they match the expected signature instead!

Reply:
Very good point - I had copied the MixFile format which used the file extension check. Fixed.